src/main/java/org/mitre/synthea/world/agents/PayerManager.java

```diff
@@ -253,59 +246,54 @@ private static Payer csvLineToPayer(Map<String, String> line) {
+    if (planId < 0) {
+      throw new RuntimeException("Plan IDs must be non-negative. Given Id " + planId + ".");
     }
-
-    String planId = line.remove(PLAN_ID).trim();
     String planName = line.remove(NAME).trim();
     Set<String> servicesCovered
         = commaSeparatedStringToHashSet(line.remove(SERVICES_COVERED).trim());
-    double deductible = Double.parseDouble(line.remove(DEDUCTIBLE).trim());
-    double defaultCoinsurance = Double.parseDouble(line.remove(COINSURANCE).trim());
-    double defaultCopay = Double.parseDouble(line.remove(COPAY).trim());
-    double monthlyPremium = Double.parseDouble(line.remove(MONTHLY_PREMIUM).trim());
+    BigDecimal deductible = new BigDecimal(line.remove(DEDUCTIBLE).trim());
+    BigDecimal defaultCoinsurance = new BigDecimal(line.remove(COINSURANCE).trim());
+    BigDecimal defaultCopay = new BigDecimal(line.remove(COPAY).trim());
+    BigDecimal monthlyPremium = new BigDecimal(line.remove(MONTHLY_PREMIUM).trim());
     boolean medicareSupplement = Boolean.parseBoolean(line.remove(MEDICARE_SUPPLEMENT).trim());
-    int yearStart = Integer.parseInt(line.remove(START_YEAR).trim());
+    boolean isACA = Boolean.parseBoolean(line.remove(ACA).trim());
+    boolean incomeBasedPremium = Boolean.parseBoolean(line.remove(INCOME_BASED_PREMIUM).trim());
+    String yearStartStr = line.remove(START_YEAR).trim();
+    int yearStart = yearStartStr.equals("") ? 0 : Integer.parseInt(yearStartStr);
     String yearEndStr = line.remove(END_YEAR).trim();
-    int yearEnd = Utilities.getYear(System.currentTimeMillis()) + 1;
-    if (!StringUtils.isBlank(yearEndStr)) {
-      yearEnd = Integer.parseInt(yearEndStr);
-    }
+    int yearEnd = StringUtils.isBlank(yearEndStr)
+        ? endYear + 1 : Integer.parseInt(yearEndStr);
+    BigDecimal maxOutOfPocket = new BigDecimal(line.remove(MAX_OOP).trim());
+    // If the priority is blank, give it minimum priority (maximum int value).
+    String priorityString = line.remove(PRIORITY_LEVEL).trim();
+    int priority = StringUtils.isBlank(priorityString)
+        ? Integer.MAX_VALUE : Integer.parseInt(priorityString);
     String eligibilityName = line.remove(ELIGIBILITY_POLICY);
 
-    payer.createPlan(servicesCovered, deductible, defaultCoinsurance,
-        defaultCopay, monthlyPremium, medicareSupplement, yearStart, yearEnd, eligibilityName);
-  }
-
-  private static Payer getPayerById(String payerId) {
-    List<Payer> payerList = getAllPayers().stream().filter(payer ->
-        payer.getPlanLinkId().equals(payerId)).collect(Collectors.toList());
-    if (payerList.size() == 1) {
-      return payerList.get(0);
-    }
-    if (payerList.size() > 1) {
-      throw new RuntimeException(payerList.size()
-          + " payers have id '" + payerId + "'. Ids should be unique.");
+    if (!PayerManager.payers.containsKey(payerId)) {
+      // Return without an error, because the given payer might only exist in another state.
+      return;
```

Comment:
This should probably be further up immediately after line 250.

If the payerId doesn't exist in the list of payers, why bother creating a half dozen BigDecimal objects and other parsing between there and here. Fail fast.